rskj-core/src/test/java/co/rsk/peg/PegUtilsGetTransactionTypeTest.java

```diff
@@ -0,0 +1,2278 @@
+                tbdActivations,
+                true,
+                false,
+                true,
+                true,
+                PegTxType.UNKNOWN
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("unknown_args")
+    void test_tx_sending_funds_to_unknown_address(
+        ActivationConfig.ForBlock activations,
+        boolean shouldUsePegoutTxIndex,
+        boolean shouldSendAmountBelowMinimum,
+        boolean existsRetiringFederation,
+        boolean existsRetiredFederation,
+        PegTxType expectedPegTxType
+    ) {
+        // Arrange
+        if (existsRetiredFederation){
+            when(provider.getLastRetiredFederationP2SHScript()).thenReturn(Optional.ofNullable(retiredFed.getP2SHScript()));
+        }
+
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+
+        Coin minimumPeginTxValue = bridgeMainnetConstants.getMinimumPeginTxValue(activations);
+        Coin belowMinimumPeginTxValue = minimumPeginTxValue.minus(Coin.SATOSHI);
+        btcTransaction.addOutput(shouldSendAmountBelowMinimum? belowMinimumPeginTxValue: minimumPeginTxValue, userAddress);
+
+        // Act
+        PegTxType pegTxType = PegUtils.getTransactionType(
+            activations,
+            provider,
+            bridgeMainnetConstants,
+            activeFederation,
+            existsRetiringFederation? retiringFederation: null,
+            btcTransaction,
+            shouldUsePegoutTxIndex? blockNumberToStartUsingPegoutIndex : 0
+        );
+
+        // Assert
+        Assertions.assertEquals(expectedPegTxType, pegTxType);
+    }
+
+    // Pegin tests
+    @Test
+    void test_pegin_below_minimum_active_fed() {
+        // Arrange
+        Federation activeFederation = bridgeMainnetConstants.getGenesisFederation();
+
+        Coin minimumPeginTxValue = bridgeMainnetConstants.getMinimumPeginTxValue(activations);
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTransaction.addOutput(minimumPeginTxValue.minus(Coin.SATOSHI), activeFederation.getAddress());
+
+        // Act
+        PegTxType pegTxType = PegUtils.getTransactionType(
+            activations,
+            provider,
+            bridgeMainnetConstants,
+            activeFederation,
+            null,
+            btcTransaction,
+            blockNumberToStartUsingPegoutIndex
+        );
+
+        // Assert
+        Assertions.assertEquals(PegTxType.PEGIN, pegTxType);
+    }
+
+    private static Stream<Arguments> pegin_args() {
+        ActivationConfig.ForBlock fingerrootActivations  = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        ActivationConfig.ForBlock tbdActivations = ActivationConfigsForTest.tbd600().forBlock(0);
```

Comment:
Shouldn't we also be testing the case where the transaction is created during the grace period after TBD600 activation?

Reply:
Test with peg-in v1 transactions?